java-client/session/src/main/java/io/deephaven/client/impl/FieldInfo.java

```diff
@@ -0,0 +1,46 @@
+package io.deephaven.client.impl;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public final class FieldInfo {
+
+    private final io.deephaven.proto.backplane.grpc.FieldInfo fieldInfo;
+
+    FieldInfo(io.deephaven.proto.backplane.grpc.FieldInfo fieldInfo) {
+        this.fieldInfo = Objects.requireNonNull(fieldInfo);
+    }
+
+    public Optional<String> type() {
+        if (fieldInfo.getTicket().getType().isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(fieldInfo.getTicket().getType());
+    }
+
+    public TicketId ticket() {
+        return new TicketId(fieldInfo.getTicket().toByteArray());
```

Comment:
I think this is wrong, it probably should be `fieldInfo.getTicket().getTicket().toByteArray()`, based only on the additional nesting added by TypedTicket. That might also be wrong, I can't tell that ticket() is actually used in a real way, it might be that `fieldInfo.getTicket().getTicket().getTicket()`(!!!) if TicketId took a ByteString, or `fieldInfo.getTicket().getTicket().getTicket().toByteArray()` (this `ByteString.toByteArray()` being different from `AbstractMessageLite.toByteArray`...).

I noticed this after not liking the js api code I was coming up with. Two notes:

 * I don't care for the name right now of this field - FieldInfo.getTicket().getTicket() seems confusing.
 * Dead horse: Do we have no tests for java-client/*, esp example/cli contents?

If I were writing this:
 * FieldInfo.ticket should be typedTicket or something like that instead, or maybe we need to embed ticket/type as two fields - since TypedTicket only happens in two places, it seems like this might be easier?
 * (follow-up) TicketId probably should take a ByteString, or a Ticket in the ctor, so as to make the mess above a bit cleaner
 * (follow-up) Tests to validate the changes to application/console fields/tickets

Reply:
Good catch. I've prefixed `typed_` in the few spots, and added Ticket as ctor of TicketId, definitely cleaner.

The java client testing is lacking outside of Batch - in particular, with console exec or application list fields. Added https://github.com/deephaven/deephaven-core/issues/1850